src/history.js

```diff
@@ -297,6 +312,39 @@ class History extends Emitter {
 
     this.size = size
   }
+
+  /**
+   * Determine if provided action is within active history branch
+   * @param {Action} action
+   */
+  isActive(action) {
+    let cursor = action
+    while (cursor) {
+      if (cursor === this.head) {
+        return true
+      }
+      cursor = cursor.next
+    }
+
+    return false
+  }
+
+  /**
+   * Starting with the provided action, navigate up the parent chain
+   * until you find an action which is active. That action is the shared
+   * root between the provided action and the current head.
+   * @param {Action} action
+   */
+  sharedRoot(action) {
+    let cursor = action
+
+    while (cursor) {
+      if (this.isActive(cursor)) {
+        return cursor
+      }
+      cursor = cursor.parent
+    }
+  }
```

Comment:
not sure if we have a pattern in place for internal or private methods? `isActive` and `sharedRoot` just exist to help abstract functionality out to help keep the public methods cleaner